mmaction/models/backbones/c3d.py

```diff
@@ -0,0 +1,136 @@
+from mmcv.cnn import ConvModule, constant_init, kaiming_init, normal_init
+from mmcv.runner import load_checkpoint
+from mmcv.utils import _BatchNorm
+from torch import nn as nn
+
+from ...utils import get_root_logger
+from ..registry import BACKBONES
+
+
+@BACKBONES.register_module()
+class C3D(nn.Module):
+    """C3D backbone.
+
+    Args:
+        pretrained (str | None): Name of pretrained model.
+        style (str): ``pytorch`` or ``caffe``. If set to "pytorch", the
+            stride-two layer is the 3x3 conv layer, otherwise the stride-two
+            layer is the first 1x1 conv layer. Default: 'pytorch'.
+        conv_cfg (dict): Config dict for convolution layer.
+            Default: ``dict(type='Conv3d')``.
+        norm_cfg (dict): Config for norm layers. required keys are ``type``,
+            Default: None.
+        act_cfg (dict): Config dict for activation layer.
+            Default: ``dict(type='ReLU')``.
+        dropout_ratio (float): Probability of dropout layer. Default: 0.5.
+        init_std (float): Std value for Initiation of fc layers. Default: 0.01.
+    """
+
+    def __init__(self,
+                 pretrained=None,
+                 style='pytorch',
+                 conv_cfg=dict(type='Conv3d'),
```

Comment:
do not use dict as default argument https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html#:~:text=Usually%20when%20a%20programmer%20uses,that%20the%20function%20is%20called.&text=The%20first%20time%20that%20the,for%20the%20list%20or%20dictionary.

Reply:
Yeah, it is not good to use mutable variable as default argument. But there is no better way to correct it currently and it is used in many places in our Open-MMLab repo. Maybe we can keep it until we find a better way to replace it.

Reply:
how does other repos handle this, like opencv, pytorch, etc?

Reply:
In pytorch, when need to use mutable variable as default arg. They set the arg as None.

and then
```
if arg is None:
    arg = [...]
```

Reply:
two solutions:
1. use the pattern in the above link, i.e., use None as default, and assign it a dict in the function body
2. use current style, but protect the dict by deepcopy and adding a note: `self.act_cfg = cp.deepcopy(act_cfg)  # protect default argument`